src/par_iter/test.rs

```diff
@@ -180,6 +180,53 @@ pub fn check_increment() {
 }
 
 #[test]
+pub fn check_skip() {
+    let a: Vec<usize> = (0..1024).collect();
+
+    let mut v1 = Vec::new();
+    a.par_iter().skip(16).collect_into(&mut v1);
+    let v2 = a.iter().skip(16).collect::<Vec<_>>();
+    assert_eq!(v1, v2);
+
+    let mut v1 = Vec::new();
+    a.par_iter().skip(2048).collect_into(&mut v1);
+    let v2 = a.iter().skip(2048).collect::<Vec<_>>();
+    assert_eq!(v1, v2);
+
+    let mut v1 = Vec::new();
+    a.par_iter().skip(0).collect_into(&mut v1);
+    let v2 = a.iter().skip(0).collect::<Vec<_>>();
+    assert_eq!(v1, v2);
+
+    // Check that the skipped elements side effects are executed
+    use std::sync::{Arc, Mutex};
+    let num = Arc::new(Mutex::new(0usize));
+    a.par_iter().map(|&n| *num.lock().unwrap() += n).skip(512).count();
```

Comment:
`Ordering::Relaxed` is fine, since this atomic not related to any other memory values.